test/Polly.Core.Tests/CircuitBreaker/Controller/CircuitStateControllerTests.cs

```diff
@@ -303,6 +303,43 @@ public async Task OnActionFailureAsync_EnsureCorrectBehavior(CircuitState state,
         }
     }
 
+    [Fact]
+    public async Task OnActionFailureAsync_EnsureBreakDurationGeneration()
+    {
+        // arrange
+        Func<ValueTask<TimeSpan>> result = () => new ValueTask<TimeSpan>(new TimeSpan(0, 0, 10, 0));
+        using var controller = CreateController(new()
+        {
+            FailureRatio = 0,
+            MinimumThroughput = 0,
+            SamplingDuration = default,
+            BreakDuration = TimeSpan.FromMinutes(1),
+            BreakDurationGenerator = static args => new ValueTask<TimeSpan>(TimeSpan.FromMinutes(args.FailureCount)),
+            OnClosed = null,
+            OnOpened = null,
+            OnHalfOpened = null,
+            ManualControl = null,
+            StateProvider = null
+        });
+
+        await TransitionToState(controller, CircuitState.Closed);
+#pragma warning disable CA2012
+        var utcNow = DateTimeOffset.MaxValue - result().GetAwaiter().GetResult();
+#pragma warning restore CA2012
```

Comment:
result isn't used for anything but this code, which is then over-complicated for essentially just taking 10 minutes off `MaxValue`.

```suggestion
        var utcNow = DateTimeOffset.MaxValue - TimeSpan.FromMinutes(10);
```